src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1175,7 +1176,8 @@ size_t QgsOgrProvider::layerCount() const
  */
 QgsWkbTypes::Type QgsOgrProvider::wkbType() const
 {
-  return static_cast<QgsWkbTypes::Type>( mOGRGeomType );
+  QgsWkbTypes::Type wkb = static_cast<QgsWkbTypes::Type>( mOGRGeomType );
+  return ogrDriverName == QLatin1String( "ESRI Shapefile" ) ? QgsWkbTypes::multiType( wkb ) : wkb;
```

Comment:
@rouault , out of curiosity, what is(are?) the historical reason(s) why shapefile advertise polygon when it really is multipolygon support?